File: objc/Visitor/Visitor/AveragePageCalcVisitor.m
#import "AveragePageCalcVisitor.h"

@implementation AveragePageCalcVisitor {
    int totalPage;
    double avgPage;
}

-(instancetype)init {
    if(self = ([super init])) {
        totalPage = 0;
        avgPage = 0.0;
    }
    return self;
}

-(double)getAvgPage {
    return avgPage;
}

-(void)visitBook:(Book*)book {
    totalPage += book.noPages;
}

-(void)visitLibrary:(Library*)library {
    avgPage = (double)totalPage / library.bookList.count;
}

@end

File: swift/Visitor/Visitor/AveragePageCalcVisitor.swift
import Foundation

class AveragePageCalcVisitor : LibraryVisitor {
    var totalPage: Int = 0
    var avgPage: Double = 0.0
    
    func getAvgPage() -> Double {
        return avgPage
    }
    
    func visitBook(_ book: Book) {
        totalPage += book.noPages
    }
    
    func visitLibrary(_ library: Library) {
        avgPage = Double(totalPage) / Double(library.bookList.count)
    }
}
